zcash_test_vectors/transaction.py

```diff
@@ -462,6 +478,9 @@ def __init__(self, rand, consensus_branch_id):
             # v^balanceSapling is defined to be 0.
             self.valueBalanceSapling = 0
 
+    def init_orchard(self, rand, flip_coins_result, is_coinbase):
+        have_orchard = (flip_coins_result >> 3) % 2
+
         # Orchard Transaction Fields
         self.vActionsOrchard = []
         if have_orchard:
```

Comment:
It's invalid to have 0 actions but a non-zero valueBalanceOrchard, for example, but the next line allows that. Change it as follows:
```python
for _ in range(max(1, rand.u8() % 5):
```
There is a 1 in 10 chance, for each test vector, that the old code would have created a tx that is invalid in this way. So the test vectors after that point might be disrupted (because the extra call to `OrchardActionDescription` will mutate `rand`). Technically we could avoid this but don't worry about it if it happens.